bundles/org.eclipse.e4.emf.xpath/src/org/eclipse/e4/emf/xpath/XPathContextFactory.java

```diff
@@ -19,7 +19,11 @@
  * Factory responsible to create an XPath-Context
  *
  * @param <T> the object type the XPath is created for
+ * @deprecated To query an E4-model use
+ *             {@link org.eclipse.e4.ui.workbench.modeling.EModelService#findMatchingElements(org.eclipse.e4.ui.model.application.MApplicationElement, String, Class)}
+ *             instead.
  */
+@Deprecated(forRemoval = true, since = "2025-03 (removal in 2027-03 or later)")
```

Comment:
I don't think we need any deprecation here if we deprecate the bundle

Reply:
I removed the additional deprecation of some methods, but the explicit deprecation of the classes should be kept as one otherwise doesn't get a corresponding warning in the code. And from my experience warnings in Manifests are overlooked more easily than in code.
Furthermore the deprecation javadoc contains additional information.

Reply:
> but the explicit deprecation of the classes should be kept as one otherwise doesn't get a corresponding warning in the code. 

You should already get a warning in the Manifest about the deprecation.